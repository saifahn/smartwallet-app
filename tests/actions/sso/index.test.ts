import { ssoActions, registrationActions } from 'src/actions'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { configure } from 'enzyme'


describe('SSO action creators', () => {

  describe('cancelSSO', () => {
    it('should clear credential request', async () => {
      const mockStore = configureStore([thunk])({})
      const action = ssoActions.cancelSSO()

      action(mockStore.dispatch)
      expect(mockStore.getActions()).toMatchSnapshot()
    })
  })

    describe('cancelReceiving', () => {
      it('should reset the selected claim', async () => {
        const mockStore = configureStore([thunk])({})
        const action = ssoActions.cancelReceiving()

        action(mockStore.dispatch)
        expect(mockStore.getActions()).toMatchSnapshot()
      })
    })
})