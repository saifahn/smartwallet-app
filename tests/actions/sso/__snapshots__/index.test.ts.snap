// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SSO action creators cancelReceiving should reset the selected claim 1`] = `
Array [
  Object {
    "type": "RESET_SELECTED",
  },
  Object {
    "actions": Array [
      Object {
        "routeName": "Home",
        "type": "Navigation/NAVIGATE",
      },
    ],
    "index": 0,
    "key": undefined,
    "type": "Navigation/RESET",
  },
]
`;

exports[`SSO action creators cancelSSO should clear credential request 1`] = `
Array [
  Object {
    "type": "CLEAR_CREDENTIAL_REQUEST",
  },
  Object {
    "actions": Array [
      Object {
        "routeName": "Home",
        "type": "Navigation/NAVIGATE",
      },
    ],
    "index": 0,
    "key": undefined,
    "type": "Navigation/RESET",
  },
]
`;
